app/config_testing.go

```diff
@@ -55,44 +60,20 @@ var evmosActivators = map[string]func(*vm.JumpTable){
 	"evmos_2": eips.Enable0002,
 }
 
-// setBaseDenomWithChainID registers the display denom and base denom and sets the
+// setBaseDenom registers the display denom and base denom and sets the
 // base denom for the chain. The function registers different values based on
-// the chainID to allow different configurations in mainnet and testnet.
-func setBaseDenomWithChainID(chainID string) error {
-	if utils.IsTestnet(chainID) {
-		return setTestnetBaseDenom()
-	}
-	return setMainnetBaseDenom()
-}
-
-func setTestnetBaseDenom() (err error) {
+// the EvmCoinInfo to allow different configurations in mainnet and testnet.
+func setBaseDenom(ci evmtypes.EvmCoinInfo) (err error) {
 	// Defer setting the base denom, and capture any potential error from it.
 	// So when failing because the denom was already registered, we ignore it and set
 	// the corresponding denom to be base denom
 	defer func() {
-		err = sdk.SetBaseDenom(types.BaseDenomTestnet)
+		err = sdk.SetBaseDenom(ci.Denom)
```

Comment:
Why is this different than `app/config.go` and we have a defer?

Reply:
> 	// Defer setting the base denom, and capture any potential error from it.
	// So when failing because the denom was already registered, we ignore it and set
	// the corresponding denom to be base denom